[Bug]: Android debug build broken

### What happened?

The debug build is broken on Android because of an `#[cfg]` mismatch between a `use` statement and the code that uses it.

### Version

latest

### In which Hotham component are you seeing the problem on?

Core

### What VR System are you seeing the problem on?

_No response_

### What OS are you seeing the problem on?

_No response_

### Relevant log output

```shell
jmgao@stormcrow:~/Projects/v3$ cargo apk b
   Compiling hotham v0.1.1 (/home/jmgao/Projects/v3/hotham/hotham)
error[E0433]: failed to resolve: use of undeclared type `DebugUtilsObjectNameInfoEXT`
    --> hotham/hotham/src/resources/vulkan_context.rs:1013:19
     |
1013 |                 &*DebugUtilsObjectNameInfoEXT::builder()
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
     |
help: consider importing one of these items
     |
1    | use ash::vk::DebugUtilsObjectNameInfoEXT;
     |
1    | use crate::vk::DebugUtilsObjectNameInfoEXT;
     |

For more information about this error, try `rustc --explain E0433`.
error: could not compile `hotham` due to previous error
Error: Command `cargo rustc --target aarch64-linux-android` had a non-zero exit code.
```

--- a/hotham-simulator/src/simulator.rs
+++ b/hotham-simulator/src/simulator.rs
@@ -394,7 +394,7 @@ pub unsafe extern "system" fn create_session(
     *session = Session::from_raw(42);
     let mut state = STATE.lock().unwrap();
     if state.device.is_none() {
-        let graphics_binding: &GraphicsBindingVulkanKHR = transmute(&(*create_info).next);
+        let graphics_binding = &*((*create_info).next as *const GraphicsBindingVulkanKHR);
         let vk_device = graphics_binding.device;
         let instance = state.vulkan_instance.as_ref().unwrap();
         let device = ash::Device::load(instance.fp_v1_0(), transmute(vk_device));

--- a/hotham/src/resources/vulkan_context.rs
+++ b/hotham/src/resources/vulkan_context.rs
@@ -1020,7 +1020,7 @@ impl VulkanContext {
 }
 
 #[allow(unused_variables)]
-fn add_device_extension_names(extension_names: &mut Vec<CString>) {
+fn add_device_extension_names(extension_names: &mut [CString]) {
     // Add Multiview extension
     #[cfg(target_os = "android")]
     extension_names.push(CString::new("VK_KHR_multiview").unwrap());

--- a/hotham/src/resources/vulkan_context.rs
+++ b/hotham/src/resources/vulkan_context.rs
@@ -19,7 +19,7 @@ use std::{cmp::max, ffi::CString, fmt::Debug, ptr::copy};
 
 type XrVulkan = xr::Vulkan;
 
-#[cfg(not(target_os = "android"))]
+#[cfg(debug_assertions)]
 use ash::vk::DebugUtilsObjectNameInfoEXT;
 
 #[derive(Clone)]

